### PR TITLE
Case-insensitive arguments for `date_trunc()`

### PR DESCRIPTION
**Pull request by yariks5s**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allows to use the `date_trunc()` function with the first argument not depending on the case of it.
Both cases are now supported: `SELECT date_trunc('day', now())` and `SELECT date_trunc('DAY', now())`

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 71277892cf453b1fb37093182d7c91254a21596b with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57624/71277892cf453b1fb37093182d7c91254a21596b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7130694329">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57624/71277892cf453b1fb37093182d7c91254a21596b/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by novikd — APPROVED** [review]
LGTM

**Comment by novikd** [comment]
Please add a simple test for it.

**Review by alexey-milovidov — APPROVED** [review]

File: tests/queries/0_stateless/02935_date_trunc_case_unsensitiveness.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+SELECT dateTrunc('DAY', toDateTime('2022-03-01 12:55:55'));
```
Comment (rschu1ze) [inline tests/queries/0_stateless/02935_date_trunc_case_unsensitiveness.sql:1]: @yariks5s Please update the [docs](https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions#date_trunc) of `date_trunc`

**Pull request merged**